core/commonMain/src/kotlinx/serialization/internal/NothingSerialDescriptor.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("FunctionName")
+@file:OptIn(ExperimentalSerializationApi::class)
+
+package kotlinx.serialization.internal
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.SerialKind
+import kotlinx.serialization.descriptors.StructureKind
+
+public object NothingSerialDescriptor : SerialDescriptor {
```

Comment:
Serialization doesn't expose its internal types.

Instead, the type should be internal and the only new public signature should be `fun NothingSerialDescriptor(): SerialDescriptor`